code/game/machinery/cryo.dm

```diff
@@ -76,113 +109,136 @@
 
 	if(abs(temperature_archived-air_contents.temperature) > 1)
 		network.update = 1
+		update_icon()
 
 	return 1
 
 /obj/machinery/atmospherics/unary/cryo_cell/relaymove(mob/user as mob)
-	if(user != occupant) //Put this in because apparently if you open the ui and press a button, you're counted as user and can trigger relaymove, this checks if you're inside, and if you aren't it ignores you for the purposes of the relaymove. - Risingvaliant
-		return
-	if(user.stat)
-		return
-	go_out()
-	return
+	if(src.occupant == user && !user.stat)
+		go_out()
 
 /obj/machinery/atmospherics/unary/cryo_cell/attack_hand(mob/user)
 	ui_interact(user)
 
-/**
- * The ui_interact proc is used to open and update Nano UIs
- * If ui_interact is not used then the UI will not update correctly
- * ui_interact is currently defined for /atom/movable (which is inherited by /obj and /mob)
- *
- * @param user /mob The mob who is interacting with this ui
- * @param ui_key string A string key to use for this ui. Allows for multiple unique uis on one obj/mob (defaut value "main")
- * @param ui /datum/nanoui This parameter is passed by the nanoui process() proc when updating an open ui
- *
- * @return nothing
- */
-/obj/machinery/atmospherics/unary/cryo_cell/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
-
-	if(user == occupant || user.stat)
-		return
+/obj/machinery/atmospherics/unary/cryo_cell/ui_interact(mob/user, datum/tgui/ui = null)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "CryoTube", "Cryo Cell")
+		ui.open()
 
-	// this is the data which will be sent to the ui
+/obj/machinery/atmospherics/unary/cryo_cell/ui_data(mob/user)
 	var/list/data = list()
 	data["isOperating"] = on
-	data["hasOccupant"] = occupant ? 1 : 0
+	data["hasOccupant"] = occupant ? TRUE : FALSE
 
 	var/list/occupantData = list()
-	if (occupant)
+	if(occupant)
 		occupantData["name"] = occupant.name
-		occupantData["stat"] = occupant.stat
-		occupantData["bodyTemperature"] = occupant.bodytemperature
-		occupantData["brain_activity"] = occupant.get_brain_status()
-		occupantData["pulse"] = occupant.get_pulse(GETPULSE_TOOL)
-		occupantData["blood_o2"] = occupant.get_blood_oxygenation()
-		occupantData["blood_pressure"] = occupant.get_blood_pressure()
-		occupantData["cloneLoss"] = occupant.getCloneLoss()
+		var/displayed_stat = occupant.stat
+		var/blood_oxygenation = occupant.get_blood_oxygenation()
+		if(occupant.status_flags & FAKEDEATH)
+			displayed_stat = DEAD
+			blood_oxygenation = min(blood_oxygenation, BLOOD_VOLUME_SURVIVE)
+		var/pulse_result
+		if(occupant.should_have_organ(BP_HEART))
+			var/obj/item/organ/internal/heart/heart = occupant.internal_organs_by_name[BP_HEART]
+			if(!heart)
+				pulse_result = 0
+			else if(BP_IS_ROBOTIC(heart))
+				pulse_result = -2
```

Comment:
This looks like a magic number, what is a negative pulse supposed to indicate?

Reply:
No idea, this was copied from the body scanner code